4337 featureuif create a new map formatting and pins updated

manav2modi opened the pull request:
## Description
_Please provide a summary of the pull request and the issue it fixes. Please add necessary details, context, dependencies, explanation of when review is needed (see next section), etc._

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [ ] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

mihail-varbanov reviewed (changes requested):
### Common

- Always check for mounted state when accessing build context after sync operations.

- Do not use explicit TextStyles, use references to styles.json (Styles().textStyles.getTextStyle()).

- Acknowledge images and icons from styles.json. Example: Icons.location_pin => Styles().images.getImage('location');

- Please create entries in app resources for strings that you use and use Localization service for accessing them, like we all do.

- Provide as large as possible tappable area for smaller UI controls like share / address details in . Make the vertical spacing between them also tappable.

- Log you changes in corresponding CHANGELOG.md


### StoriedSightsBottomSheet

- You should create a more complex model for filters. Just think about what would happen when you will have to localize the filter labels.

- Taps on Place's address row should launch directions to that place, see Explore.launchDirections().

- You should find a way to link the Place model to the existing Explore model so that you could reuse all existing infrastructure for this. Example: Explore.launchDirections().

- If StoriedSightsBottomSheet is not going to be reused in other places it should be moved to explore directory and have appropriate name for that location.

### ExploreMapPanel

- What is (or should be) the relationship between the existing MapPanel and StoriedSightsBottomSheet? For example, why you fill mental health buildings when Storied Sites are selected?

mihail-varbanov commented:
Hi @manav2modi,

Yesterday I was able to look at [UIF - Campus Destinations jam.pdf](https://drive.google.com/file/d/1X9jTsRyFCemhedYyCrLq8EH2MZJSi-TX/view?usp=drive_web) and got an idea what you are trying to achieve. If I had to do this I would do the following:

1\. Implement [Explore](https://github.com/rokwire/app-flutter-plugin/blob/support/v1.7/lib/model/explore.dart#L22) interface in  [Place](https://github.com/rokwire/app-flutter-plugin/blob/support/v1.7/lib/model/places.dart) class. An example for this is the same [implementation](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/model/StudentCourse.dart#L332-L348) in [Building](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/model/StudentCourse.dart#L204) class. You could also handle Place Explores types in [ExploreExt](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ext/Explore.dart), [ExploreCard](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/explore/ExploreCard.dart) and [ExploreDetailPanel](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/explore/ExploreDetailPanel.dart) for completeness. 

2\. Threat ExploreMapType.StoriedSites in [ExploreMapPanel](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/explore/ExploreMapPanel.dart) in exactly the same way as the other enum entries. When "Storied Sites" is selected make a call to [Places.​​getAllPlaces](https://github.com/rokwire/app-flutter-plugin/blob/support/v1.7/lib/service/places.dart#L17), supply the result as list of explore items and let the existing ExploreMapPanel's implementation fill the markers of the storied sites.

When ExploreMapType.StoriedSites is selected **and** [Places.​​getAllPlaces](https://github.com/rokwire/app-flutter-plugin/blob/support/v1.7/lib/service/places.dart#L17) has completed, show [StoriedSightsBottomSheet](https://docs.google.com/document/d/12nCPiR5Fc2K_fSwJB3IyKgNPKVLXLCpHivIHJRhVLSc/edit) inside ExploreMapPanel in the same way as now but pass the retrieved stored places as a parameter to its constructor. Acknowledge this list in [_StoriedSightsBottomSheetState](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/widgets/StoriedSightsBottomSheet.dart#L16C7-L16C37) to display the stored places.

3\. Keep [_StoriedSightsBottomSheetState._selectedFilters](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/widgets/StoriedSightsBottomSheet.dart#L19) content in [Storage](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/service/Storage.dart) service in the same maner as [excludedFoodIngredientsPrefs](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/service/Storage.dart#L52-L57) or [includedFoodTypesPrefs](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/service/Storage.dart#L59-L64). When the user updates the selected filters selection just store the new selection in [Storage](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/service/Storage.dart) service and fire a notification using NotificationService. Listen for this notification in [ExploreMapPanel](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/explore/ExploreMapPanel.dart) and when it comes [refresh](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/explore/ExploreMapPanel.dart#L1837) the stored places list applying the new selection stored in [Storage](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/service/Storage.dart) service. An example of similar processing in ExploreMapPanel is [Event2FilterParam.notifyChanged](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/events2/Event2HomePanel.dart#L1113) or [StudentCourses.notifySelectedTermChanged](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/service/StudentCourses.dart#L29).

4\. It was not defined explicitly in [UIF - Campus Destinations jam.pdf](https://drive.google.com/file/d/1X9jTsRyFCemhedYyCrLq8EH2MZJSi-TX/view?usp=drive_web) what should happen on taping on Stored Place map marker, but I think the common sense shows that it should get selected in [StoriedSightsBottomSheet](https://docs.google.com/document/d/12nCPiR5Fc2K_fSwJB3IyKgNPKVLXLCpHivIHJRhVLSc/edit), instead of [selecting](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/explore/ExploreMapPanel.dart#L689) it in the ExploreMapPanel like other marker types do. There are different ways to do this, my preference is to invoke staightly [_StoriedSightsBottomSheetState._onDestinationTap](https://github.com/rokwire/illinois-app/blob/4337-featureuif-create-a-new-map-formatting-and-pins-updated/lib/ui/widgets/StoriedSightsBottomSheet.dart#L582) from ExporeMapPanel. You can access StoriedSightsBottomSheet's state from ExploreMapPanel if you assign it a key of GlobalKey<StoriedSightsBottomSheet> type. An example how parent widget can access child widget's states can be found [here](https://github.com/rokwire/illinois-app/blob/develop/lib/ui/safety/SafetySafeWalkRequestPage.dart#L238-L246) (please note that this exists only in the current [develop](https://github.com/rokwire/illinois-app/tree/develop) branch and not in the [support/v6.0](https://github.com/rokwire/illinois-app/tree/support/v6.0) branch).

Let me know please if you have any questions or if there is anything you need some help with.

manav2modi commented:
Hey! @mihail-varbanov Thanks for the review. I have fixed most of the concerns and problems you mentioned and still fixing a few more. Let me know if you see anything else that might need to be done.

Also we won't be doing the point number 3 as we don't plan on storing what filters a person has selected.

manav2modi commented:
Hey! @mihail-varbanov. I have addressed most of the issues you have mentioned. I still have some stuff that is in todo that I will be implementing early next week.
8.1, 8.2 and 7, along with some more minor changes are left to be implemented

mihail-varbanov commented:
Hi @manav2modi,

You have made a good progress but there are still issues from the previous lists to resolve:

2\. Branch conflicts are still not resolved:
<img width="724" alt="Screenshot 2024-10-21 at 11 00 18 AM" src="https://github.com/user-attachments/assets/d930a51f-3ffe-4f8e-af7b-4301281fadfc">

4\. Your selection handling looks good. I would only recommend to clear `_selectedFilters` each time you apply some external selection to `_storiedSights` in `selectPlace` or `selectPlaces` methods. Otherwise it becomes inconsistent: there are some filters selected but they are not applied to `_storiedSights` display list.

8\. All items need resolution, incl. 8.3. Let me know please if there is something unclear in these items.

8\.4\. In addition, please fix the newly introduced warnings:
```
misho@MacBookPro2 illinois-app % dart analyze
Analyzing illinois-app...              14.6s

warning • lib/ui/explore/ExploreStoriedSightsBottomSheet.dart:41:8 • The
          value of the field '_isLightboxVisible' isn't used. Try removing
          the field, or using it. • unused_field
warning • lib/ui/explore/ExploreStoriedSightsBottomSheet.dart:41:8 • The
          value of the field '_isLightboxVisible' isn't used. Try removing
          the field, or using it. • unused_field
warning • lib/ui/explore/ExploreStoriedSightsBottomSheet.dart:42:23 • The
          value of the field '_selectedImage' isn't used. Try removing the
          field, or using it. • unused_field
warning • lib/ui/explore/ExploreStoriedSightsBottomSheet.dart:42:23 • The
          value of the field '_selectedImage' isn't used. Try removing the
          field, or using it. • unused_field

4 issues found.
```
9\. This is a new issue: implement Places deep links handling (it is marked as TBD in your comment).

Please resolve all known issues, create a new PR for merging your work in [support/v6.0](https://github.com/rokwire/illinois-app/tree/support/v6.0), and let me know to make the review (hopefully the final one).

manav2modi commented:
Hey @mihail-varbanov I have fixed all the issues mentioned. (Hoping this is the last review for this as well). A few things to note:
- I am currently adding the deeplinking stuff to the places and have a PR open on the plugin (https://github.com/rokwire/app-flutter-plugin/pull/492) I will be another PR that will be dependent on this branch to merge additional changes needed for deeplinking. Decided to do a different PR as to make it clear what changes are needed to get the deeplinking working on the illinois app.
- I was not sure why I would need to open another PR as all the changes are here. If you still want me to let me know and I can do that but for now I have pushed everything here.
- I was not sure what else needed to be done for 8.3 I made sure to make it so it is easily tappable and have not faced any issues regarding the same
-
